sdk/python/kfp/components/_python_op.py

```diff
@@ -56,7 +56,18 @@ def _capture_function_code_using_cloudpickle(func) -> str:
     finally:
         sys.modules[func.__module__] = old_module
     func_code = '{func_name} = pickle.loads({func_pickle})'.format(func_name=func.__name__, func_pickle=repr(func_pickle))
-    return 'import pickle' + '\n\n' + func_code
+
+    code_lines = [
+        'import subprocess',
+        'import sys',
+        'subprocess.call([sys.executable, "-m", "pip", "install", "cloudpickle"])'
```

Comment:
Is the lightweight component using the python engine instead of python3? If so, should we use python3 and here also pip3?

Reply:
> If so, should we use python3 and here also pip3?

This line does that.

`pip3` is the same as
```python3 -m pip install my_package```
But the latter one is better because in some environments `python3` and `pip3` can point to different environments. I've seen people having the issues. So `python3 -m pip install my_package` is more reliable. And `sys.executable` ensures that the same executable (`python3`) gets used.